interface/super/edit_globals.php

```diff
@@ -34,24 +34,26 @@
 require_once("$srcdir/classes/CouchDB.class.php");
 require_once(dirname(__FILE__)."/../../myportal/soap_service/portal_connectivity.php");
 
-if ($_GET['mode'] != "user") {
+$userMode = (array_key_exists('mode', $_GET) && $_GET['mode'] == 'user');
+
+if (!$userMode) {
   // Check authorization.
   $thisauth = acl_check('admin', 'super');
   if (!$thisauth) die(xlt('Not authorized'));
 }
 
 function checkCreateCDB(){
-  $globalsres = sqlStatement("SELECT gl_name, gl_index, gl_value FROM globals WHERE gl_name IN 
+  $globalsres = sqlStatement("SELECT gl_name, gl_index, gl_value FROM globals WHERE gl_name IN
   ('couchdb_host','couchdb_user','couchdb_pass','couchdb_port','couchdb_dbase','document_storage_method')");
     $options = array();
     while($globalsrow = sqlFetchArray($globalsres)){
       $GLOBALS[$globalsrow['gl_name']] = $globalsrow['gl_value'];
     }
     $directory_created = false;
-  if($GLOBALS['document_storage_method'] != 0){
+  if(array_key_exists('document_storage_method', $GLOBALS) && $GLOBALS['document_storage_method'] != 0){
```

Comment:
This one could be replaced with **empty**, decided to go for consistency and verbosity.